src/main/java/org/broadinstitute/dsm/model/BSPKit.java

```diff
@@ -0,0 +1,157 @@
+import org.broadinstitute.dsm.model.bsp.BSPKitInfo;
+import org.broadinstitute.dsm.model.bsp.BSPKitStatus;
+import org.broadinstitute.dsm.statics.ApplicationConfigConstants;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class BSPKit {
+    private static Logger logger = LoggerFactory.getLogger(BSPKit.class);
+
+    public Optional<BSPKitStatus> getKitStatus(@NonNull String kitLabel, NotificationUtil notificationUtil) {
+        logger.info("Checking label " + kitLabel);
+        Optional<BSPKitQueryResultDto> bspKitQueryResult = BSPKitQueryResultDao.getBSPKitQueryResult(kitLabel);
+        Optional<BSPKitStatus> result = Optional.empty();
+        if (bspKitQueryResult.isEmpty()) {
+            logger.info("No kit w/ label " + kitLabel + " found");
+        }
+        else {
+            BSPKitQueryResultDto maybeBspKitQueryResult = bspKitQueryResult.get();
+            if (StringUtils.isNotBlank(maybeBspKitQueryResult.getParticipantExitId())) {
+                String message = "Kit of exited participant " + maybeBspKitQueryResult.getBspParticipantId() + " was received by GP.<br>";
+                notificationUtil.sentNotification(maybeBspKitQueryResult.getNotificationRecipient(), message, NotificationUtil.DSM_SUBJECT);
+                result = Optional.of(new BSPKitStatus(BSPKitStatus.EXITED));
+            }
+            else if (StringUtils.isNotBlank(maybeBspKitQueryResult.getDeactivationDate())) {
+                result = Optional.of(new BSPKitStatus(BSPKitStatus.DEACTIVATED));
+            }
+        }
+        return result;
+    }
+
+    public boolean canReceiveKit(@NonNull String kitLabel) {
+        Optional<BSPKitQueryResultDto> bspKitQueryResult = BSPKitQueryResultDao.getBSPKitQueryResult(kitLabel);
+        if (bspKitQueryResult.isEmpty()) {
+            logger.info("No kit w/ label " + kitLabel + " found");
+            return false;
+        }
+        else {
+            BSPKitQueryResultDto maybeBspKitQueryResult = bspKitQueryResult.get();
+            if (StringUtils.isNotBlank(maybeBspKitQueryResult.getParticipantExitId()) || StringUtils.isNotBlank(maybeBspKitQueryResult.getDeactivationDate())) {
+                logger.info("Kit can not be received.");
+                return false;
+            }
+        }
+        return true;
+
+    }
+
+    public Optional<BSPKitInfo> receiveBSPKit(String kitLabel, NotificationUtil notificationUtil) {
+        logger.info("Trying to receive kit " + kitLabel);
+        return TransactionWrapper.inTransaction(conn -> {
+            Optional<BSPKitQueryResultDto> bspKitQueryResult = BSPKitQueryResultDao.getBSPKitQueryResult(kitLabel);
+             BSPKitQueryResultDto maybeBspKitQueryResult = bspKitQueryResult.get();
```

Comment:
calling `.get()` method of `Optional` without checking if it presents, may throw `NoSuchElementException`, use either `isPresent` or `ifPresent`.